applications/CompressiblePotentialFlowApplication/custom_elements/incompressible_potential_flow_element.cpp

```diff
@@ -420,18 +423,20 @@ void IncompressiblePotentialFlowElement<Dim, NumNodes>::CalculateLocalSystemWake
     // Calculate shape functions
     GeometryUtils::CalculateGeometryData(GetGeometry(), data.DN_DX, data.N, data.vol);
 
+    const double density_infinity = 1.0; //TODO: Read from rCurrentProcessInfo[DENSITY_INFINITY] once available
+
     GetWakeDistances(data.distances);
 
-    Matrix lhs_total = ZeroMatrix(NumNodes, NumNodes);
+    BoundedMatrix<double, NumNodes, NumNodes> lhs_total = ZeroMatrix(NumNodes, NumNodes);
 
-    ComputeLHSGaussPointContribution(data.vol, lhs_total, data);
+    ComputeLHSGaussPointContribution(data.vol*density_infinity, lhs_total, data);
 
     if (this->Is(STRUCTURE))
     {
-        Matrix lhs_positive = ZeroMatrix(NumNodes, NumNodes);
-        Matrix lhs_negative = ZeroMatrix(NumNodes, NumNodes);
+        BoundedMatrix<double, NumNodes, NumNodes> lhs_positive = ZeroMatrix(NumNodes, NumNodes);
+        BoundedMatrix<double, NumNodes, NumNodes> lhs_negative = ZeroMatrix(NumNodes, NumNodes);
 
-        CalculateLocalSystemSubdividedElement(lhs_positive, lhs_negative);
+        CalculateLocalSystemSubdividedElement(lhs_positive, lhs_negative, rCurrentProcessInfo);
         AssignLocalSystemSubdividedElement(rLeftHandSideMatrix, lhs_positive,
```

Comment:
I'm wondering if you can save the allocation of `lhs_positive` and `lhs_negative` by creating a unique function that directly adds their contributions to the LHS.

Reply:
I see what you mean. I will think how to do what you say and open another PR with this change.